### PR TITLE
Cirrus: Guarantee $DEST_BRANCH is passed through

### PR DESCRIPTION
**Pull request by cevich**
There are several runtime contexts (rootless and container) where
`$DEST_BRANCH` is needed but was not supplied.  A prior commit
(c4865767171b) removed the default value, `main` which was being
set incorrectly when CI ran on release branches.  Fix this by ensuring
the variable is non-empty upon entry to `setup_environment.sh`, then
ensure it gets passed through to child environments by way of the
`/etc/ci_environment` file.  This will maintain compatibility with
both CI and `hack/get_ci_vm.sh` use.

Also Ref: https://github.com/containers/podman/issues/14131

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@edsantiago PTAL
/approve

**Review by vrothberg — APPROVED** [review]
LGTM

Let's wait for Ed's blessing.

**Comment by edsantiago** [comment]
The reason I didn't do this in my PR is: https://github.com/containers/podman/pull/14179/files#diff-f3866c83e5fdf3650df2ad5b9c0c18efaa71891d23675732cc931e90fd2b10faL54-R57

As I read this, it looks like a chicken-egg problem: those variables need to be already set in order to be written out; and, didn't we resolve that DEST_BRANCH is unset in cron and/or branch context?

**Review by edsantiago — CHANGES_REQUESTED** [review]
I'd like to understand the cron/branch question before merging

**Comment by edsantiago** [comment]
>link isn't working like I think you intended.

Ugh. Thanks, github. That was supposed to link to the 'while ... <passthrough >> /etc/ci_environment` lines, and I think I've answered the question I had (re: chicken/egg relating to how `ci_environment` gets written).

>Does that help answer your questions?

Yes, in the sense that I trust you've given it the thought it needs, thanks. So:

/approve
/lgtm

**Comment by cevich** [comment]
Thanks @edsantiago, I appreciate the scrutiny.

**Pull request merged**
